### PR TITLE
DAT-19308 - feat(databricks): ensure essential table properties Delta.*

### PR DESCRIPTION
**Pull request by CharlesQueiroz**
## Overview
This PR addresses the handling of essential Delta table properties in the Databricks extension, specifically ensuring critical properties are always included in table creation statements regardless of user input.

### Problem
Previously, when users specified custom TBLPROPERTIES during table creation, the system would not guarantee the presence of essential Delta properties. 

### Solution
The implementation introduces a property management system that:

1. Defines a set of essential Delta properties that must always be present
2. Provides default values for these properties
3. Implements property merging logic that:
   - Preserves user-specified values for essential properties when provided
   - Automatically adds missing essential properties with default values
   - Maintains any additional custom properties specified by users

### Key Changes
- Created constants for essential properties and their default values
- Implemented property processing logic in `processTableProperties` method
- Added test coverage for various property combinations
- Ensured backward compatibility with existing table creation workflows

### Testing
The changes are covered by unit tests that verify:
- Default property inclusion
- Custom property preservation
- Mixed property scenarios
- Various value types (strings, numbers, booleans)
- Edge cases with empty or null properties

### CODE REVIEW

File: src/main/java/liquibase/ext/databricks/sqlgenerator/CreateTableGeneratorDatabricks.java [hunk 1]
```diff
@@ -1,22 +1,43 @@
...
+            "'delta.enableChangeDataFeed'"
+    );
+
+    private static final Map<String, String> DEFAULT_VALUES = Map.of(
+            "'delta.columnMapping.mode'", "'name'",
+            "'delta.enableDeletionVectors'", "true",
+            "'delta.feature.allowColumnDefaults'", "'supported'",
+            "'delta.logRetentionDuration'", "'30 days'",
+            "'delta.deletedFileRetentionDuration'", "'7 days'",
+            "'delta.targetFileSize'", "134217728",
```
Comment (filipelautert) [inline src/main/java/liquibase/ext/databricks/sqlgenerator/CreateTableGeneratorDatabricks.java:37]: Where did those values came from? Do we need them?
Cause before we only had 3 columns with default values: 
* 'delta.feature.allowColumnDefaults' = 'supported'
* 'delta.columnMapping.mode' = 'name'
* 'delta.enableDeletionVectors' = true

**Review by filipelautert — CHANGES_REQUESTED** [review]

File: src/main/java/liquibase/ext/databricks/sqlgenerator/CreateTableGeneratorDatabricks.java [hunk 2]
```diff
@@ -1,22 +1,36 @@
...
 import org.apache.commons.lang3.StringUtils;
 
-import java.util.List;
+import java.util.*;
+
+import static java.util.stream.Collectors.joining;
 
 public class CreateTableGeneratorDatabricks extends CreateTableGenerator {
 
+    private static final Set<String> ESSENTIAL_PROPERTIES = Set.of(
```
Comment (filipelautert) [inline src/main/java/liquibase/ext/databricks/sqlgenerator/CreateTableGeneratorDatabricks.java:22]: Could you use the map keys (like in `DEFAULT_VALUES.keySet() `) instead of having a `ESSENTIAL_PROPERTIES` with the same values?

**Review by filipelautert — CHANGES_REQUESTED** [review]

**Review by filipelautert — APPROVED** [review]

**Review by SvampX — APPROVED** [review]
Looks good to me. 
Had a concern about wrapping of properties key/value in quotes, but acccording to tests logic uses existing user specified approach and default values are wrapped.

**Review by SvampX — CHANGES_REQUESTED** [review]
@CharlesQueiroz we need to update expected SQL-s, see failed harness tests.

**Review by SvampX — APPROVED** [review]

File: pom.xml [hunk 3]
```diff
@@ -66,7 +66,7 @@
         <liquibase-test-harness.version>1.0.10</liquibase-test-harness.version>
         <junit.version>5.11.4</junit.version>
         <junit-platform.version>1.11.4</junit-platform.version>
-        <mockito.version>5.15.2</mockito.version>
+        <mockito.version>4.11.0</mockito.version>
```
Comment (KushnirykOleh) [inline pom.xml:69]: extension should work on java 8, never mockito requires java 11
